src/riscv-integration.adoc

```diff
@@ -993,10 +1027,25 @@ xref:stval-format[xrefstyle=short] to assist software in handling the trap.
 [#stval-format]
 include::img/stvalreg.edn[]
 
-TYPE is a CHERI-specific fault type that caused the exception while CAUSE
-is the cause of the fault. The possible CHERI types and causes are encoded as
-shown in xref:mtval-cheri-type[xrefstyle=short] and
-xref:mtval-cheri-causes[xrefstyle=short] respectively.
+The fields are identical to <<mtval>> for CHERI exceptions.
+
+[#stval2,reftext="stval2"]
+==== Supervisor Trap Value Register 2 (stval2)
+
+The <<stval2>> register is an SXLEN-bit read-write register.
+
+The CSR address is 0x24b.
+
+When a fault is taken into S-mode <<stval2>> is updated as for <<mtval2>> for all CHERI memory access traps.
```

Comment:
The mtval2 case says this:
> mtval2 is set to zero for all PCC faults

Which is a little different to saying that its only updated for CHERI memory access faults. It would be good to be a little more precise in the intention for both of these registers (I was trying to hint at this in an earlier comments although I did not do it very well!)

Reply:
clarified